src/actions/promote.ts

```diff
@@ -25,13 +26,13 @@ export async function promote() {
         await promoteCanary(kubectl);
     } else if (isBlueGreenDeploymentStrategy()) {
         await promoteBlueGreen(kubectl);
+        // Adding traceability data
+        const deploymentConfig = await getDeploymentConfig();
+        await AzureTraceabilityHelper.addTraceability(kubectl, deploymentConfig);
     } else {
         core.debug('Strategy is not canary or blue-green deployment. Invalid request.');
         throw ('InvalidPromotetActionDeploymentStrategy');
     }
-
-    // Adding traceability data
-    await AzureTraceabilityHelper.addTraceability(kubectl);

```

Comment:
Why move this code?

Reply:
This will already be called inside the if() block, where deploymentHelper.deploy() is called.
So I moved it to the else if() block to prevent repetition.

Reply:
good catch! This was missed earlier.